server/chat-plugins/smogtours.ts

```diff
@@ -0,0 +1,428 @@
+			const idx = section.tours.findIndex(t => t.id === tourID);
+			const title = section.tours[idx].title;
+			if (idx < 0) {
+				return this.popupReply(`Tour with ID "${tourID}" not found.`);
+			}
+			section.tours.splice(idx, 1);
+			this.refreshPage(`tournament-view-${sectionID}-${tourID}`);
+			this.popupReply(`Tour "${title}" ended.`);
+		},
+		whitelist(target, room, user) {
+			this.checkCan('rangeban');
+			const [sectionID, targetID] = target.split(',').map(toID).filter(Boolean);
+			if (!sectionID || !targetID) {
+				return this.parse(`/help smogtours`);
+			}
+			const section = tours[sectionID];
+			if (!section) {
+				return this.errorReply(`Invalid section ID: "${sectionID}". Valid IDs: ${Object.keys(tours).join(', ')}`);
+			}
+			if (section.whitelist?.includes(targetID)) {
+				return this.errorReply(`That user is already whitelisted on that section.`);
+			}
+			if (!section.whitelist) section.whitelist = [];
+			section.whitelist.push(targetID);
+			this.privateGlobalModAction(
+				`${user.name} whitelisted ${targetID} to manage tours for the ${section.title} section`
+			);
+			this.globalModlog('TOUR WHITELIST', targetID);
+			saveTours();
+		},
+		unwhitelist(target, room, user) {
+			this.checkCan('rangeban');
+			const [sectionID, targetID] = target.split(',').map(toID).filter(Boolean);
+			if (!sectionID || !targetID) {
+				return this.parse(`/help smogtours`);
+			}
+			const section = tours[sectionID];
+			if (!section) {
+				return this.errorReply(`Invalid section ID: "${sectionID}". Valid IDs: ${Object.keys(tours).join(', ')}`);
+			}
+			const idx = section.whitelist?.indexOf(targetID);
+			if (!section.whitelist || !idx || idx < 0) {
+				return this.errorReply(`${targetID} is not whitelisted in that section.`);
+			}
+			section.whitelist?.splice(idx, 1);
+			if (!section.whitelist.length) {
+				delete section.whitelist;
+			}
+			this.privateGlobalModAction(
+				`${user.name} removed ${targetID} from the tour management whitelist for the ${section.title} section`
+			);
+			this.globalModlog('TOUR UNWHITELIST', targetID);
+			saveTours();
+		},
+		view() {
+			return this.parse(`/join view-tournaments-all`);
+		},
+	},
+	smogtourshelp: [
+		`/smogtours view - View a list of ongoing forum tournaments.`,
+		`/smogtours whitelist [section], [user] - Whitelists the given [user] to manage tournaments for the given [section].`,
+		`Requires: &`,
+		`/smogtours unwhitelist [section], [user] - Removes the given [user] from the [section]'s management whitelist.`,
+		`Requires: &`,
+	],
+};
+
+function renderTab(inner: string, isTitle?: boolean, isCur?: boolean) {
+	isTitle = false;
+	let buf = '';
+	if (isCur) {
+		buf += `<div class="folder cur"><div class="folderhack3"><div class="folderhack1">`;
+		buf += `</div><div class="folderhack2"></div>`;
+		buf += `<div class="selectFolder">${inner}</div></div></div>`;
+	} else {
+		if (!isTitle) {
+			inner = `<div class="selectFolder">${inner}</div>`;
```

Comment:
you should add a jsdoc comment to this function to explain that it mutates `inner`